cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -26,6 +26,14 @@
 
 #include <string>
 
+// Define a new macro analogous to CUDA_HOST_DEVICE_CALLABLE that maximizes the
+// chances of inlining when applied to a callable that is called on the GPU.
+#ifdef __CUDACC__
+#define CUDA_HOST_DEVICE_FORCEINLINE_CALLABLE __host__ __device__ __forceinline__
```

Comment:
Maybe we can do something that's less of a mouthful? Something abbreviated like `CUDF_HDFI`

Reply:
TBH I kinda prefer the long name in this case. It's only being used once, and to me that infrequent use means that 1) it's not too painful since it's very rarely typed anyway, and 2) it's immediately obvious what it means to readers who have never encountered it.

Reply:
In this instance I'm also inclined to prefer the long name because IMO the abbreviation would make it pretty difficult to find or understand. But I can change it if you have a strong preference.

Reply:
I'll grant that `HDFI` will be obscure the first time you see it, but I think it would be one of those things where you learn it once and remember it forever.

Reply:
@bdice @jrhemstad So it seems like our options are

- `CUDA_HOST_DEVICE_FORCEINLINE_CALLABLE`: Very long and leads to bad formatting, but also very descriptive and clear for users who probably won't see it often.
- `CUDA_HDFI`: Short and concise, but obscure and very different looking from its close counterpart `CUDA_HOST_DEVICE_CALLABLE`.
- `CUDA_HOST_DEVICE_FORCEINLINE`: Still pretty long, but now it's similar in length to `CUDA_HOST_DEVICE_CALLABLE` and is still very descriptive, although there is some cognitive dissonance between the two because of the missing `CALLABLE` in one case.
- Alternatives?

I'm not convinced that people will necessarily get used to the abbreviated version since it will probably be used far less frequently than the non-force inline versions, and it's hard to know where to find the definition of a macro like this when everything includes types.hpp, but I'm fine being voted down.

Cast your votes now!! May the best name win.

Reply:
My vote is `CUDA_HOST_DEVICE_FORCEINLINE_CALLABLE`.

If I may use a ranked-choice ballot, I say:
1. `CUDA_HOST_DEVICE_FORCEINLINE_CALLABLE`
2. `CUDA_HOST_DEVICE_FORCEINLINE` (less symmetry, but `CALLABLE` is arguably unnecessary in all these macros)
3. `CUDA_HDFI` (too unclear)

I looked at the formatting in @jrhemstad's [comment](https://github.com/rapidsai/cudf/pull/9530#discussion_r742130231) but I don't have any issue with it. It just pushed the arguments of the function onto a new line. I might have even preferred it, honestly -- because there aren't two lines for `Functor f` and `Ts&&... args` with lots of leading whitespace.

Reply:
https://github.com/rapidsai/raft/blob/6c93d46a1f3e17e613497f510baa7e449b7abe59/cpp/include/raft/cuda_utils.cuh#L38-L41

RAFT already uses the same convention as I suggested.

Reply:
My claims above were assuming that the suggestion to shorten was only being applied to these new macros, and not the preexisting ones. @jrhemstad clarified offline that he intended for both, which I'm on board with since people will get used to it pretty quickly if the normal (non forceinline) versions are also changed. I'll limit the scope of this PR to just these macros, but I'll rename all of them project-wide in a subsequent PR.

Reply:
I suggest dropping "CALLABLE" because it's easily inferred that this is a function modifier.  So I guess I'm agreeing with @bdice. I don't like "DI", "HDFI", etc.